src/Ocelot.Provider.Consul/Consul.cs

```diff
@@ -30,9 +30,8 @@ public virtual async Task<List<Service>> GetAsync()
         var nodesTask = _consul.Catalog.Nodes();
 
         await Task.WhenAll(entriesTask, nodesTask);
-
-        var entries = entriesTask.Result.Response ?? Array.Empty<ServiceEntry>();
-        var nodes = nodesTask.Result.Response ?? Array.Empty<Node>();
+        var entries = (await entriesTask).Response ?? Array.Empty<ServiceEntry>();
+        var nodes = (await nodesTask).Response ?? Array.Empty<Node>();
```

Comment:
Will we await 3 times?
The goal of `Task.WhenAll` is parallelization!

Reply:
I've double checked the MS docs and both variants are good
- we can read already initialized `Result` property because the task was completed by `Task.WhenAll`
- we can await twice with `await task`, no new execution because of task completion (see [remarks](https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.task-1.result?view=net-8.0#remarks)), and read the result from implicit local variable in expression `(await task).prop`

The 2nd statement has better readability, that's why **there is no issue.**